src/README.md

```diff
@@ -1,34 +1,67 @@
-# NOTE
-The files in this directory (including this README.md) is imported from [DPSL](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/).
+# NOTE#1
+The files in this directory (including this README.md) is a subset of [DPSL doc](https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/).
+
+# Note#2
+You may refer to [cros-diag-app/diagnostics-extension](https://github.com/MahmoudAGawad/cros-diag-app/tree/main/diagnostics-extension) to see the integration of this library in action.
 
 # Overview
 Diagnostic Processor Support Library (DPSL) is a collection of telemetry and
-diagnostics interfaces exposed to third-parties:
+diagnostics interfaces exposed to third-parties. All API functions are
+accessed through the dpsl.* namespace:
    - dpsl.diagnostics
     | Diagnostics interface for running device diagnostics routines (tests).
    - dpsl.telemetry
     | Telemetry (a.k.a. Probe) interface for getting device telemetry
     | information.
-  - dpsl.system_events
-    | System Events interface for (un)subscribing to system events.
+
+# Supported APIs/Functions
+Currently, the following is supported:
+- dpsl.telemetry:
+-- dpsl.telemetry.getVpdInfo(): Promise<dpsl.VpdInfo>     // see types.js
+-- dpsl.telemetry.getOemData(): Promise<dpsl.OemDataInfo>
+- dpsl.diagnostics:
+-- dpsl.diagnostics.getAvailableRoutines(): Promise<dpsl.AvailableRoutinesList>
+-- dpsl.diagnostics.battery.runCapacityRoutine(): Promise<Routine>
+-- dpsl.diagnostics.battery.runHealthRoutine(): Promise<Routine>
+-- dpsl.diagnostics.battery.runDischargeRoutine(params: dpsl.BatteryDischargeRoutineParams): Promise<Routine>
+-- dpsl.diagnostics.battery.runChargeRoutine(params: dpsl.BatteryChargeRoutineParams): Promise<Routine>
+-- dpsl.diagnostics.cpu.runCacheRoutine(params: dpsl.CpuRoutineDurationParams): Promise<Routine>
+-- dpsl.diagnostics.cpu.runStressRoutine(params: dpsl.CpuRoutineDurationParams): Promise<Routine>
+-- dpsl.diagnostics.memory.runMemoryRoutine(): Promise<Routine>
+
+## Dealing with diagnostic routines
```

Comment:
nit: maybe we should mention that after each reboot ID will start from 0 and our partners might need to remember returned ID for long running routines to be able to get status if SW will be terminated after timeou